arch/arm/core/swap.S

```diff
@@ -324,6 +341,41 @@ _oops:
     blx _do_kernel_oops
     pop {pc}
 
+#if CONFIG_USERSPACE
+    /*
+     * System call will setup a jump to the _do_arm_syscall function
+     * when the SVC returns via the bx lr.
+     *
+     * There is some trickery involved here because we have to preserve
+     * the original LR value so that we can return back to the caller of
+     * the SVC.
+     *
+     * On SVC exeption, the stack looks like the following:
+     * r0 - r1 - r2 - r3 - r12 - LR - PC - PSR
+     * r5 - r6 - call id - saved LR
+     *
+     */
+_do_syscall:
+    ldr r1, [r0, #24]   /* grab address of PC from stack frame */
+    str r1, [r0, #44]   /* store address to use for LR after syscall */
+    ldr r1,=_arm_do_syscall
+    str r1, [r0, #24]   /* overwrite the LR to point to _arm_do_syscall */
+
+    /* validate syscall limit, only set priv mode if valid */
+    ldr ip,=_SYSCALL_LIMIT
+    ldr r1, [r0, #40]
+    cmp r1,ip
+    bge skip_raising_privilege
```

Comment:
I think we still need to raise privilege even the id is over _SYSCALL_LIMIT.
Because we still need the privilege to execute _handler_bad_syscall() later.

Reply:
I've tested the syscall bad path and it appears to execute fine in user mode and prints out the error messages just fine.

Reply:
yeah, I forget that when we do printk in _handler_bad_syscall(), the privilege will be raised if we are in user mode.
It seems it won't hurt if we always do raise prilivege here and it can also reduce conditional branch.